### DISCUSSION TITLE
How to include ReceivableSpecifiedTradeAccountingAccount (BT-133)

### DISCUSSION THREAD

**Discussion by Alb-Creator** [stephanstapel/ZUGFeRD-csharp#289] [Q&A] [answered]
Hi,
I have a problem with defining and inserting ReceivableSpecifiedTradeAccountingAccount (BT-133) into a TradeLineItem.
my code:

![Screenshot 2024-07-01 181147](https://github.com/stephanstapel/ZUGFeRD-csharp/assets/174350015/d043c643-63a4-457c-8093-9efefb0004ee)

I don't know how to do it. I would be very happy to receive a code snippet.
Thanks

**Comment by stephanstapel** [comment]
You might use this one:
https://github.com/stephanstapel/ZUGFeRD-csharp/blob/2caafc1a519a7cee6a696d5501f2dd3db7e5d986/ZUGFeRD/TradeLineItem.cs#L264

If possible, could you share your use cases? It would be really interesting to understand how this element is used in practice.

**Comment by Alb-Creator** [comment]
Thanks for your answer.
Unfortunately, I don't know how to add this information. TradeLineItem does not contain a variable for this. buyerAssignedID corresponds to BT-156
![Screenshot 2024-07-02 091358](https://github.com/stephanstapel/ZUGFeRD-csharp/assets/174350015/95863ea2-af4a-442d-986b-95355b8a037d)

Our case is that a school authority orders textbooks for different schools/classes. If each item is for a different school/class, the accounting department can use this field to automatically assign the costs. This information is supplied with the order. So they expect us to provide this information in the xRechnung.

**Reply by stephanstapel** [reply, marked as answer]
thanks for your explanation!
AddTradeLineItem() returns the TradeLineItem object. You might call the AddReceivableSpecifiedTradeAccountingAccount() on this object.

**Comment by Alb-Creator** [comment]
Thank you!
That was the solution. Now it works.
-------------------------------------------
"As soon as you do it right, it works" :-)
